pkg/apis/v1/nodepool.go

```diff
@@ -326,7 +332,7 @@ func (in *NodePool) MustGetAllowedDisruptions(ctx context.Context, c clock.Clock
 func (in *NodePool) GetAllowedDisruptionsByReason(ctx context.Context, c clock.Clock, numNodes int) (map[DisruptionReason]int, error) {
 	var multiErr error
 	allowedDisruptions := map[DisruptionReason]int{}
-	for _, reason := range WellKnownDisruptionReasons {
+	for reason := range WellKnownDisruptionReasons {
```

Comment:
I wonder if we even need a definition of WellKnownDisruptionReasons? Is it possible to say that if there are no constraints that are returned by this function that it's just the default? e.g. Unlimited?

Reply:
I just made this as a passed in variable, so that we can define it as needed